libraries/joomla/database/driver/mysqli.php

```diff
@@ -557,8 +557,7 @@ public function execute()
 
 		if (!is_object($this->connection))
 		{
-			JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database');
-			throw new RuntimeException($this->errorMsg, $this->errorNum);
+			throw new RuntimeException('Could not connect to MySQL.');
```

Comment:
Actually, connect() *may* fail without it suffering from a logic error but simply from a server restriction. Let's say you have the following timeline:
* T+0.00 connect() is fired
* T+0.20 a query is fired
* T+1.30 a second query is fired. However MySQL has gone away because it closes the connection after 1 second.
* T+1.30 connect() is fired again. However it fails because the host has blocked multiple connections coming from the same process (yes, I have seen that)

It is not that common but it *can* happen without implying a logic error in connect().